apps/imageHotspotCreator/src/components/Creator/createHotspot.tsx

```diff
@@ -123,65 +125,63 @@ const CreateHotspot = ({
     const scaledWidth = imageWidth * scale;
     const scaledHeight = imageHeight * scale;
 
-    canvas.width = scaledWidth;
+   if(canvas){
+     canvas.width = scaledWidth;
     canvas.height = scaledHeight;
     canvas.style.cursor = cursorStyle;
     context.drawImage(image, 0, 0, scaledWidth, scaledHeight);
     setCanvas(canvas);
 
     // Update hotspot arrays based on the new canvas size
     if (sdk.entry.fields.hotspots.getValue()?.hotspots) {
+
       setRectArray(sdk.entry.fields.hotspots.getValue().hotspots)
       setListArray(sdk.entry.fields.hotspots.getValue().hotspots)
+
     } else {
+
       sdk.entry.fields.hotspots.setValue({
         hotspots: [],
       })
+
       setRectArray(sdk?.entry?.fields?.hotspots?.getValue().hotspots)
       setListArray(sdk?.entry?.fields?.hotspots?.getValue().hotspots)
     }
+  }
+
+
   };
 
+  const [missedField, setMissedField] = useState([]);
+
+  useEffect(() => {
+
+    const requiredNames = ["Title", "Image URL", "Hotspots"];
+    const missedObject: any = cloneDeep(missedField);
+
+    // Check if required names exist and create objects accordingly
+    for (const name of requiredNames) {
+      const found = sdk.contentType.fields.find((obj: any) => obj.name === name); // Check if object with name exists
+      if (!found) {
+        missedObject.push({ name }); // Add missing object with just the name property
+      }
+    }
+    if (missedObject.length > 0) {
+      setMissedField(missedObject);
+    }
+
+  }, [])
```

Comment:
Is it intentional that the dependency array is empty?

Reply:
Yes, It will happen only once initially.